File: 04-quick-sort/ruby/recursive_sum_test.rb
# frozen_string_literal: true

require 'minitest/autorun'
require_relative './recursive_sum'

class RecursiveSumTest < Minitest::Test
  def test_recursive_sum_returns_an_integer
    arr = [1, 2, 3, 4]
    assert_instance_of Integer, recursive_sum(arr)
  end

  def test_recursive_sum_returns_sum_of_all_items
    arr = [1, 2, 3, 4]
    assert_equal 10, recursive_sum(arr)
  end

  def test_recursive_sum_returns_zero_if_list_is_empty
    assert_equal 0, recursive_sum([])
  end
end

File: 04-quick-sort/python/recursive_sum_test.py
import unittest
from .recursive_sum import recursive_sum

class TestRecursiveSum(unittest.TestCase):
    def setUp(self):
        pass

    def test_recursive_sum_returns_a_number(self):
        arr = [1, 2, 3, 4]
        self.assertIsInstance(recursive_sum(arr), int)

    def test_recursive_sum_returns_sum_of_all_items(self):
        arr = [1, 2, 3, 4]
        self.assertEqual(recursive_sum(arr), 10)

    def test_recursive_sum_returns_zero_if_list_is_empty(self):
        self.assertEqual(recursive_sum([]), 0)
